tests/src/Interop/MarshalAPI/MarshalStructure/MarshalStructure.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        VerifyByValBoolArray();
+        VerifyByValArrayInStruct();
+        VerfiyByValDateArray();
+        return 100;
+    }
+    
+    static void VerifyByValBoolArray()
+    {
+        var structure1 = new StructWithBoolArray()
+        {
+            array = new bool[]
+            {
+                true,true,true,true
+            }
+        };
+
+        int size = Marshal.SizeOf(structure1);
+        IntPtr memory = Marshal.AllocHGlobal(size);
```

Comment:
I have an idea. Why not allocate additional guard bytes here? We can write a magic number (0xf00df00d, for example) at the end, and make sure that magic number is not overwriten. It is much better than hoping for an AV.